src/compilerlib/pb_parsing_parser.mly

```diff
@@ -181,6 +190,67 @@ reserved :
   | T_reserved extension_range_list semicolon {$2}
 /* T_toDO: incomplete, reserved field can also be defined by field names */
 
+service :
+  | T_service T_ident T_lbrace service_body_content_list rbrace {
+    Pb_parsing_util.service ~content:$4 (snd $2)
+  }
+  | T_service T_ident T_lbrace rbrace {
+    Pb_parsing_util.service ~content:[]  (snd $2)
+  }
+
+service_body_content_list :
+  | service_body_content  { [$1] }
+  | service_body_content service_body_content_list { $1::$2 }
+
+service_body_content :
+  | rpc          { Pb_parsing_util.service_body_rpc $1 }
+  | option       { Pb_parsing_util.service_body_option $1 }
+
+rpc :
+  | T_rpc T_ident T_lparen T_ident T_rparen T_returns T_lparen T_ident T_rparen semicolon {
+    Pb_parsing_util.rpc ~req:(snd $4) ~res:(snd $8) (snd $2)
+  }
+  | T_rpc T_ident T_lparen T_ident T_rparen T_returns T_lparen T_ident T_rparen rpc_options {
+    Pb_parsing_util.rpc ~req:(snd $4) ~res:(snd $8) ~options:$10 (snd $2)
+  }
+
+rpc_options :
+  | T_lbrace rpc_options_list T_rbrace           { $2 };
+  | T_lbrace rpc_options_list T_rbrace semicolon { $2 };
+  | T_lbrace T_rbrace                            { Pb_option.empty };
+  | T_lbrace T_rbrace semicolon                  { Pb_option.empty };
+
+rpc_options_list :
+  | rpc_option                          {
+    let option_name, option_value = $1 in
+    Pb_option.add Pb_option.empty option_name option_value
+  }
+  | rpc_option rpc_options_list  {
+    Pb_option.add $2 (fst $1) (snd $1)
+  }
+
+rpc_option :
+  | T_option option_identifier T_equal constant semicolon { ($2, $4) }
+  | T_option option_identifier T_equal rpc_option_content semicolon { ($2, $4) }
+
+rpc_option_content :
+    /*
+     * covers [option (google.api.http) = { get: "/v1/shelves/{shelf}" };]
```

Comment:
it's not a perfect solution and eventually it will need to be changed, but the current options don't support objects

once service generation is needed (this is not likely to happen anytime soon especially since it is client dependent) perhaps this could be changed